docs/courses/robotics/3.5-computer-vision.md

```diff
@@ -0,0 +1,393 @@
+    camera_params = [fx, fy, cx, cy]
+    
+    for tag in tags:
+        # Get tag pose
+        pose, e0, e1 = tag.fit_pose(camera_params, tag_size)
+        
+        # Extract rotation and translation
+        rotation_matrix = pose[:3, :3]
+        translation_vector = pose[:3, 3]
+        
+        # Convert rotation matrix to Euler angles
+        euler_angles = cv2.Rodrigues(rotation_matrix)[0].flatten()
+        
+        # Draw tag outline
+        cv2.polylines(image, [np.int32(tag.corners)], True, (0, 255, 0), 2)
+        
+        # Draw tag center and ID
+        center = tuple(map(int, tag.center))
+        cv2.circle(image, center, 5, (0, 0, 255), -1)
+        cv2.putText(image, str(tag.tag_id), (center[0] - 10, center[1] - 10),
+                    cv2.FONT_HERSHEY_SIMPLEX, 0.5, (0, 0, 255), 2)
+        
+        # Print spatial information
+        print(f"Tag ID: {tag.tag_id}")
+        print(f"Position: X={translation_vector[0]:.2f}, Y={translation_vector[1]:.2f}, Z={translation_vector[2]:.2f}")
+        print(f"Rotation: Roll={np.degrees(euler_angles[0]):.2f}, Pitch={np.degrees(euler_angles[1]):.2f}, Yaw={np.degrees(euler_angles[2]):.2f}")
+        print("---")
+    
+    return image
+
+def main():
+    # Initialize camera
+    cap = cv2.VideoCapture(0)  # Use 0 for default camera, adjust if needed
+    
+    while True:
+        ret, frame = cap.read()
+        if not ret:
+            print("Failed to capture image")
+            break
+        
+        # Detect AprilTags
+        tags = detect_apriltags(frame)
+        
+        # Perform spatial referencing
+        frame_with_tags = spatial_referencing(frame, tags)
+        
+        # Display the result
+        cv2.imshow('AprilTag Spatial Referencing', frame_with_tags)
+        
+        if cv2.waitKey(1) & 0xFF == ord('q'):
+            break
+    
+    cap.release()
+    cv2.destroyAllWindows()
+
+if __name__ == "__main__":
+    main()
+```
+
+This example assumes you have a camera connected and AprilTags in view. You'll need to adjust the camera matrix parameters (fx, fy, cx, cy) to match your specific camera's calibration. Also, make sure to set the correct tag_size in meters.
+
+---
+
+### Motorized Microscope and OpenCV
+
+After detecting regions of interest (ROI) using **OpenCV** and **AprilTags**, you will integrate a motorized microscope. The microscope will move automatically based on the ROI found in the image, providing fine control and automation for analyzing samples.
+
+#### Demo
+
+✅ Watch [Controlling a Motorized Microscope](https://youtu.be/fHZSsAKThW4)
+
+This example code combines OpenCV image processing techniques with simulated microscope functionality to implement automated monitoring of yeast growth. The main features include:
+1. Image preprocessing: Using Gaussian blur and adaptive thresholding to enhance image quality.
+2. Yeast cell detection: Using contour detection methods to identify potential yeast cells.
```

Comment:
Would yeast cells be visible with a 40x or 100x objective lens? 

If not, this could be changed to be for other types of cells or for microspheres of some kind.

Overall I like the workflow and example.